modules/lib/models/transaction.js

```diff
@@ -16,24 +19,326 @@ module.exports = (osseus) => {
     })
   }
 
-  transaction.create = (from, to, amount) => {
+  const validateAmount = (amount) => {
+    return new Promise(async (resolve, reject) => {
+      amount = new BigNumber(amount)
+      if (amount.eq(0) || amount.lt(0)) {
+        reject(new Error(`amount must be positive`))
+      }
+      if (amount.isNaN()) {
+        reject(new Error(`amount illegal`))
+      }
+      resolve(amount)
+    })
+  }
+
+  const validateBlockchainBalances = () => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const tasks = []
+        const wallets = await osseus.db_models.wallet.getAll()
+        osseus.logger.debug(`validateBlockchainBalances - found ${wallets.length} wallets`)
+        wallets.forEach(wallet => {
+          let address = wallet.address
+          let currencies = wallet.balances.map(balance => balance.currency.ccAddress)
+          currencies.forEach(currency => {
+            tasks.push(new Promise(async (resolve, reject) => {
+              try {
+                let valid = await osseus.utils.validateBlockchainBalance(address, currency)
```

Comment:
i don't think that a call to a logic from the utils should be done from the model class
controllers (or util logic) should call the model functions, not the other way around

Reply:
it's in the `lib` so all is ok :)